README.md

```diff
@@ -190,6 +194,110 @@ pip install ni-measurement-plugin-sdk-generator
+
+        - `--module-name` and `--class-name` define the module and class names of the generated client. If not specified, they are derived from the measurement service class name.
+
+        - `--directory-out` specifies the output directory for the generated files. If not provided, files are placed in the current directory.
+        
+        > **Note**: For multiple measurement client creation, `--module-name` and `--class-name` are ignored, and they are derived from service class for each measurement. So, ensure measurement service class name adheres to proper naming conventions.
+
+    2. Run this command to create measurement clients for all registered measurements.
+
+        `ni-measurement-plugin-sdk-generator --all`
+
+        > **Note**: `--directory-out` can be provided for this mode.
+
+    3. Run this command to interactively create measurement clients for any registered measurements.
+
+        `ni-measurement-plugin-sdk-generator --interactive`
+
+3. The genarted client comprises four APIs: `measure`, `stream_measure`, `register_pin_map`, and `cancel`. The usage of these APIs is discussed in the section ["Steps to run/debug the measurement client".](#steps-to-rundebug-the-measurement-client)
+
+> **Note**:
+> - The Measurement Plug-In Client is compatible with all datatypes supported by the Measurement Plug-In.
+> - Measurement configuration of Double XY datatype is not supported. 
+> - For Enum datatypes, the generated enum class names will be the measurement parameter names suffixed with 'Enum'. For instance, if configuration is 'Enum In', the generated enum in client will be `EnumInEnum'.
+> - Ring controls in LabVIEW measurement will be represented as numeric datatypes in the generated client.
+
+### 
+
+---
+
+## Steps to run/debug the measurement client
+
+1. Make sure the required measurement service is running before interacting with it via the client.
+
+2. Use the client APIs from the ["Developing a Minimal Python MeasurementClient"](#developing-a-minimal-python-measurement-client) section.
+
+    1. For non-streaming measurements, use `measure` method.
+
+        ``` python
+        from sample_measurement.SampleMeasurement import measure
+        
+        output = measure()
+        ```
+
+    2. For streaming measurements, use `stream_measure` method.
+
+        ``` python
+        from sample_measurement.SampleMeasurement import stream_measure
+        
+        outputs = stream_measure()
+        ```
+
+    3. If measurements require pin map, pin map can be registered using `register_pin_map` method. By default, `sites` is set to 0.
+
+        ``` python
+        from sample_measurement.SampleMeasurement import measure, register_pin_map
+        
+        register_pin_map(pin_map_path)
+        output = measure()
+        ```
+        - Alternatively, you can provide pin map context through `pin_map_context` property. Similarly, sites can also be provided through `sites` property.
```

Comment:
'Alternatively, you can provide a pin map context through the `pin_map_context` property. Similarly, sites can also be provided through the `sites` property.'